Add TODO.md to Docs

wimrijnders opened the pull request:
Redo of #26 , this time branching from `development`.
Previous branched from another PR and contained too many commits.

Also added change to `.gitignore`.

-----
Previous text:

This is my complete TODO (and then some), made available for public use and scrutiny. I hope it proves useful to others.

I intend to update the `TODO.md` as I proceed. It is therefore a dynamic document.

The direct benefit for @mn416 is that he can now see what my intentions are wrt. the project in the foreseeable future. Hoping for feedback.

mn416 commented:
>  'Getting Started' to front page

I quite like the current structure: the front page advertises the library, and potential users can then follow the link to get started.  We could add another link to the guide at the bottom of the front page, just in case the reader missed it at the start.

> Convert AutoStart to unit test

Do you mean `AutoTest`?

I don't think the todo list should include things that are actually done.  If you want that, then create a separate file called `ChangeLog.md` that records all changes from the stable version.

Other than that, looks sensible.

wimrijnders commented:
I personally think the front page is too full. I can agree that having `Getting Started` on a separate page is sensible, but I find it helps to get a glimpse of the effort involved to set up the project. As in 'only four commands? That's easy!'

However, having the elaboration of the examples on the front page is overkill. I really think that they're better off in a separate page.

Yes, I mean `AutoTest`. It's a prime candidate for the unit tests, for which you'll be receiving a PR tomorrow for the initial setup.

OK, I'll remove the done-things. I just like to see that progress is being made, otherwise you only see a load of things not done yet - depressing. But I'll take them off, no problem.

wimrijnders commented:
Notes on last commit:

- Addresses the review points
- Incorporates points raised during discussion
- Adjusted grouping of points to make it more coherent
